src/web3/WalletClient.ts

```diff
@@ -328,31 +308,34 @@ export class WalletClient extends BaseClient implements IWalletClient {
 
   /**
    * Generates a new wallet account.
+   * @param version_number - The version number of the secret key to be generated, to create a new account.
    *
    * @returns A Promise that resolves to an {@link IAccount} object, which represents the newly created account.
    */
-  public static async walletGenerateNewAccount(): Promise<IAccount> {
+  public static async walletGenerateNewAccount(
+    version_number: number,
```

Comment:
Why is it an argument of the wallet generation function ? Can a user choose between version ?

Reply:
I didn't want to hardcode the VERSION_NUMBER for maintainability reasons. However when generating a new wallet, there is no information available within the code to determine the desired version, so it needs to be specified as an argument.
In the future we should make this argument optional and fetch the current version if not provided (but then this would pose a problem for offline wallet generation)

Reply:
I think you shouldn't let the choice to the caller and use the last known version instead